### PR TITLE
Structured display for adjoint/transpose

### PR DESCRIPTION
**Pull request by jishnub**
On master
```julia
julia> D = Diagonal(1:3)
3×3 Diagonal{Int64, UnitRange{Int64}}:
 1  ⋅  ⋅
 ⋅  2  ⋅
 ⋅  ⋅  3

julia> Adjoint(D)
3×3 adjoint(::Diagonal{Int64, UnitRange{Int64}}) with eltype Int64:
 1  0  0
 0  2  0
 0  0  3
```
After this PR
```julia
julia> Adjoint(D)
3×3 adjoint(::Diagonal{Int64, UnitRange{Int64}}) with eltype Int64:
 1  ⋅  ⋅
 ⋅  2  ⋅
 ⋅  ⋅  3
```

This example is somewhat artificial, but the structured printing makes it easier to work with various sparse/banded array types.

### CODE REVIEW

**Comment by jishnub** [comment]
Updated, so now we have
```julia
julia> Symmetric(Bidiagonal(1:4, 1:3, :U))
4×4 Symmetric{Int64, Bidiagonal{Int64, UnitRange{Int64}}}:
 1  1  ⋅  ⋅
 1  2  2  ⋅
 ⋅  2  3  3
 ⋅  ⋅  3  4
```

**Comment by jishnub** [comment]
Gentle bump

**Review by dkarrasch — APPROVED** [review]
You can go ahead and merge your own PR. ;-)

**Comment by jishnub** [comment]
Was waiting for an approval :)

**Pull request merged**
